File: backend/Models/Story.cs
using System.ComponentModel.DataAnnotations;

namespace backend.Models
{
    public class Story
    {
        public int Id { get; set; }
        public int CulturalSiteId { get; set; }

        [Required, MaxLength(160)]
        public string Title { get; set; } = default!;

        [Required, MaxLength(40)]
        public string StoryType { get; set; } = default!;

        [Range(1, 1000)]
        public int EstimatedReadTimeMinutes { get; set; }

        [Required, MaxLength(5000)]
        public string FullContent { get; set; } = default!;

        [MaxLength(500)]
        public string? HistoricalContext { get; set; }

        [MaxLength(500)]
        public string? CulturalSignificance { get; set; }

        public CulturalSite? CulturalSite { get; set; }
        public ICollection<StoryImage> Images { get; set; } = new List<StoryImage>();
    }

    public class StoryImage
    {
        public int Id { get; set; }
        public int StoryId { get; set; }

        [Required, MaxLength(300)]
        public string Url { get; set; } = default!;

        [Required, MaxLength(300)]
        public string PublicId { get; set; } = default!;

        public int Position { get; set; } = 0;
        public Story? Story { get; set; }
    }
}

File: frontend/lib/data/models/Story.dart
class Story {
  final int id;
  final int culturalSiteId;
  final String title;
  final String storyType;
  final int estimatedReadTimeMinutes;
  final String fullContent;
  final String? historicalContext;
  final String? culturalSignificance;
  final List<String> imageUrls;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  Story({
    required this.id,
    required this.culturalSiteId,
    required this.title,
    required this.storyType,
    required this.estimatedReadTimeMinutes,
    required this.fullContent,
    this.historicalContext,
    this.culturalSignificance,
    this.imageUrls = const [],
    this.createdAt,
    this.updatedAt,
  });

  factory Story.fromJson(Map<String, dynamic> json) {
    return Story(
      id: json['id'] as int? ?? 0,
      culturalSiteId: json['culturalSiteId'] as int? ?? 0,
      title: json['title']?.toString() ?? '',
      storyType: json['storyType']?.toString() ?? '',
      estimatedReadTimeMinutes: json['estimatedReadTimeMinutes'] as int? ?? 0,
      fullContent: json['fullContent']?.toString() ?? '',
      historicalContext: json['historicalContext']?.toString(),
      culturalSignificance: json['culturalSignificance']?.toString(),
      imageUrls: (json['imageUrls'] as List<dynamic>?)
          ?.map((e) => e.toString())
          .toList() ??
          [],
      createdAt: json['createdAt'] != null
          ? DateTime.tryParse(json['createdAt'].toString())
          : null,
      updatedAt: json['updatedAt'] != null
          ? DateTime.tryParse(json['updatedAt'].toString())
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'culturalSiteId': culturalSiteId,
      'title': title,
      'storyType': storyType,
      'estimatedReadTimeMinutes': estimatedReadTimeMinutes,
      'fullContent': fullContent,
      'historicalContext': historicalContext,
      'culturalSignificance': culturalSignificance,
      'imageUrls': imageUrls,
      'createdAt': createdAt?.toIso8601String(),
      'updatedAt': updatedAt?.toIso8601String(),
    };
  }
}